atomic_reactor/tasks/common.py

```diff
@@ -59,6 +60,7 @@ def source(self) -> source.Source:
     @classmethod
     def from_cli_args(cls, args: dict):
         """Create a TaskParams instance from CLI arguments."""
+        task_result = args.pop("task_result", None)
         args = cls._drop_known_unset_args(args)
```

Comment:
I see that `_drop_known_unset_args` will cause issues with `Optional` params. IMO we can remove the method, it's the CLI's responsibility to make sure required params will never be unset.

(and then `task_result` wouldn't need special handling here)

Reply:
well I can try to remove it, dunno if it won't break stuff

Reply:
It shouldn't, the CLI looks correct

Reply:
dropped